capybara/init.lua

```diff
@@ -0,0 +1,167 @@
+--= Capybara for Creatures MOB-Engine (cme) =--
+-- Copyright (c) 2015-2016 BlockMen <blockmen2015@gmail.com>
+--
+-- init.lua
+--
+-- This software is provided 'as-is', without any express or implied warranty. In no
+-- event will the authors be held liable for any damages arising from the use of
+-- this software.
+--
+-- Permission is granted to anyone to use this software for any purpose, including
+-- commercial applications, and to alter it and redistribute it freely, subject to the
+-- following restrictions:
+--
+-- 1. The origin of this software must not be misrepresented; you must not
+-- claim that you wrote the original software. If you use this software in a
+-- product, an acknowledgment in the product documentation is required.
+-- 2. Altered source versions must be plainly marked as such, and must not
+-- be misrepresented as being the original software.
+-- 3. This notice may not be removed or altered from any source distribution.
+
+-- Main capybara mob definition
+local def = {
+
+	-- Basic stats
+	name = "creatures:capybara",
+	stats = {
+		hp = 8,
+		lifetime = 450,
+		can_jump = 1,
+		can_swim = true,
+		can_burn = true,
+		can_panic = true,
+		has_falldamage = true,
+		has_kockback = true,
+	},
+
+	-- Model & animation setup
+	model = {
+		mesh = "capybara.b3d",
+		textures = {"capybara.png"},
+		collisionbox = {-0.3, -0.55, -0.2, 0.3, 0.3, 0.2}, 
+		rotation = 90.0,
+		animations = {
+			idle = {start = 21, stop = 21, speed = 1},
+			walk = {start = 1, stop = 20, speed = 15},
+			walk_long = {start = 1, stop = 20, speed = 15},
+		},
+	},
+
+	-- Sound definitions
+	sounds = {
+		on_damage = {name = "creatures_capybara_hit", gain = 1.0, distance = 10},
+		on_death = {name = "creatures_capybara_death", gain = 1.0, distance = 10},
+		swim = {name = "creatures_splash", gain = 1.0, distance = 10},
+		random = {
+			idle = {name = "creatures_capybara", gain = 0.6, distance = 10, time_min = 23},
+		},
+	},
+
+	-- Behavior modes
+	modes = {
+		idle = {chance = 0.5, duration = 10, update_yaw = 8},
+		walk = {chance = 0.2, duration = 4.5, moving_speed = 1.3},
+		walk_long = {chance = 0.15, duration = 8, moving_speed = 1.3, update_yaw = 5},
+
+		-- Follow behavior
+		follow = {
+			chance = 0,
+			duration = 20,
+			radius = 4,
+			timer = 5,
+			moving_speed = 1,
+			items = {"default:apple", "farming:melon_slice"}
+		},
+
+		-- Eating behavior
+		eat = {
+			chance = 0.15,
+			duration = 4,
+			nodes = {
+				"default:grass_1", "default:grass_2", "default:grass_3",
+				"default:grass_4", "default:grass_5", "default:dirt_with_grass"
+			}
+		},
+	},
+
+	-- Drops on death
+	drops = function(self)
+		if not self then
+			return
+		end
+		local pos = self.object:getpos()
```

Comment:
Deprecated. Use `get_pos`. This is also logged to debug.txt and the chat console if you use the minetest.conf setting `deprecated_lua_api_handling = log`.